kotlinx-coroutines-core/common/src/JobSupport.kt

```diff
@@ -522,7 +523,32 @@ public open class JobSupport constructor(active: Boolean) : Job, ChildJob, Paren
         return node
     }
 
-    private fun addLastAtomic(expect: Any, list: NodeList, node: JobNode) =
+    /**
+     * Adds the [node] to the given [list] if the state of this job is equal to [expect].
```

Comment:
I believe reviewing this doc is beneficial to properly review the next step -- removal of `addLastAtomic`